htlcswitch/link_test.go

```diff
@@ -1058,15 +1058,17 @@ func TestChannelLinkMultiHopUnknownNextHop(t *testing.T) {
 	htlcAmt, totalTimelock, hops := generateHops(amount, testStartingHeight,
 		n.firstBobChannelLink, n.carolChannelLink)
 
-	daveServer, err := newMockServer(
-		t, "dave", testStartingHeight, nil, n.globalPolicy.TimeLockDelta,
+	// Remove bob's outgoing link with Carol. This will cause him to fail
+	// back the payment to Alice since he is unaware of Carol when the
+	// payment comes across.
+	bobChanID := lnwire.NewChanIDFromOutPoint(
+		&channels.bobToCarol.State().FundingOutpoint,
 	)
-	if err != nil {
-		t.Fatalf("unable to init dave's server: %v", err)
-	}
-	davePub := daveServer.PubKey()
-	receiver := n.bobServer
-	rhash, err := n.makePayment(n.aliceServer, n.bobServer, davePub, hops,
+	n.bobServer.htlcSwitch.RemoveLink(bobChanID)
```

Comment:
Seems the test can stay as it was? Before it added dave "who doesn't exist in the network" such that the first outgoing link would fail.

Reply:
oh, i just assumed from the test comment "checks that we receive remote error from Bob if he
 has no idea about next hop"

Reply:
if we keep it as it was, we'd have to make another test, as no forwarding packages would be written if it gets rejected before leaving alice. this is what led my to realize the something was weird, cause there were no forwarding packages written for any of the channels